app-backend/common/src/main/scala/cache/CacheClient.scala

```diff
@@ -0,0 +1,83 @@
+package com.azavea.rf.common.cache
+
+import java.util.concurrent.Executors
+
+import net.spy.memcached._
+
+import scala.concurrent._
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import com.azavea.rf.common.Config
+import cats.data._
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.typesafe.scalalogging.LazyLogging
+
+object CacheClientThreadPool {
+  implicit lazy val ec: ExecutionContext =
+    ExecutionContext.fromExecutor(
+      Executors.newFixedThreadPool(
+        Config.memcached.threads,
+        new ThreadFactoryBuilder().setNameFormat("cache-client-%d").build()
+      )
+    )
+}
+
+class CacheClient(client: => MemcachedClient) extends LazyLogging {
+
+  import CacheClientThreadPool._
+
+  val cacheEnabled = Config.memcached.enabled
+
+  def delete(key: String) =
+    if(cacheEnabled) {
+      client.delete(key)
+    }
+
+  def setValue[T](key: String, value: T, ttlSeconds: Int = 0): Unit = {
+    Future {
+      client.set(key, ttlSeconds, value)
+    }
+  }
+
+  def getOrElseUpdate[CachedType](cacheKey: String, expensiveOperation: => Future[CachedType],
+    ttlSeconds: Int = 0, doCache: Boolean = true): Future[CachedType] = {
+
+    if (cacheEnabled && doCache) {
+
+      val futureCached = Future { client.asyncGet(cacheKey).get() }
```

Comment:
Yup definitely -- I'll add a comment. In this case that's why we added `CacheClientThreadPool` since before this function was taking an implicit `ExecutionContext` and would potentially block a thread in an EC that might not be intended. Most of this was left as-is I think except for that.